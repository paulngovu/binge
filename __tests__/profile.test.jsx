import React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react';
import Profile from '../pages/profile';

describe('Profile page', () => {
  it('renders profile elements', () => {
    const tree = renderer.create(<Profile />).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should allow name to be changed', () => {
    // render profile with Joe Bruin (default name)
    const profile = render(<Profile />);
    expect(profile.getByText('Joe Bruin')).toBeTruthy();

    // edit name
    fireEvent.click(profile.getByText(/Edit/i));
    const input = profile.getByTestId('input-name');
    fireEvent.change(input, {target: {value: 'Sally Bruin'}})
    expect(input.value).toBe('Sally Bruin');

    // submit change
    fireEvent.click(profile.getByTestId('submit-btn'));

    // expect profile to render new name instead of old name
    expect(profile.getByText('Sally Bruin')).toBeTruthy();
    expect(profile.queryByText('Joe Bruin')).toBeNull();
  })
});

