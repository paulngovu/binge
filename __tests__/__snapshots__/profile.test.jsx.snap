// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile page renders profile elements 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 hMmsfQ"
>
  <header
    className="StyledBox-sc-13pk1d4-0 gpGtYH"
    display="flex"
  >
    <h1
      className="StyledHeading-sc-1rdh4aw-0 elclaf"
    >
      Binge
    </h1>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 itCgil"
    />
    <div
      alignself="end"
    >
      <a
        className="StyledButton-sc-323bzc-0 haWZSW"
        data-testid="home-btn"
        href="/"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        <svg
          aria-label="Home"
          className="StyledIcon-sc-ofa7kd-0 bwvewX"
          viewBox="0 0 24 24"
        >
          <path
            d="m1 11 11-9 11 9m-8 12v-8H9v8m-5 0V9m16 14V9"
            fill="none"
            stroke="#000"
            strokeWidth="2"
          />
        </svg>
      </a>
    </div>
  </header>
  <div
    className="StyledBox-sc-13pk1d4-0 hHCfST"
  >
    <div
      className="jsx-840335723 container"
    >
      <img
        className="jsx-840335723 profile-img"
        src="/joe-bruin.jpg"
      />
      <div
        className="StyledBox-sc-13pk1d4-0 WOYhc"
      >
        <span
          className="StyledText-sc-1sadyjn-0 jmAKBa"
        >
          Joe Bruin
        </span>
      </div>
      <div
        className="jsx-840335723 bio-container"
      >
        <span
          className="StyledText-sc-1sadyjn-0 iumzz"
        >
          Default bio.
        </span>
      </div>
      <button
        className="StyledButton-sc-323bzc-0 hmUEoa"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        type="button"
      >
        Edit
      </button>
      <div
        className="StyledBox-sc-13pk1d4-0 gXhMXE"
      >
        <button
          className="StyledButton-sc-323bzc-0 emnQnE"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          type="button"
        >
          Log Out
        </button>
      </div>
    </div>
  </div>
</div>
`;
